### PR TITLE
Added product details page

### PR DESCRIPTION
**Pull request by sanket4120**
Click on 
- Shop Now button on home page carousel
- product card
- horizontal product card image and title 

will redirect to product details page

Files to review:
- pages (product)
- actions (product actions (getProductDetails) )
- reducers (produtReducer (productDetailsReducer) )

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -13,6 +13,8 @@ import Wishlist from './pages/wishlist/Wishlist';
 import Cart from './pages/cart/Cart';
 import AuthRequired from './components/authRequired/AuthRequired';
 import Account from './pages/account/Account';
+import Product from './pages/product/Product';
+import NotFound from './pages/notFound/NotFound';
```
Comment (shraddha15-creator) [inline src/App.js:17]: use index.js file for import export

File: src/pages/product/Product.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
+import { Link, useLocation, useNavigate, useParams } from 'react-router-dom';
+import { useEffect } from 'react';
+import { getProductDetails } from '../../actions/productActions';
+import { useProducts } from '../../context/productsContext';
+import { useUser } from '../../context/userContext';
+import { useMessage } from '../../context/messageContext';
```
Comment (shraddha15-creator) [inline src/pages/product/Product.js:6]: Here as well

**Review by shraddha15-creator — APPROVED** [review]
Code is clean

**Review by arulk8 — APPROVED** [review]
UI is eye catching. Good job

**Pull request merged**